python/ELASTICSEARCH_ENCRYPTED_AT_REST/ELASTICSEARCH_ENCRYPTED_AT_REST.py

```diff
@@ -0,0 +1,360 @@
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+
+Rule Name:
+    ELASTICSEARCH_ENCRYPTED_AT_REST
+Description:
+    Checks whether Amazon Elasticsearch Service (Amazon ES) domains have encryption at rest configuration enabled. The resource is NON_COMPLIANT if encryption at rest is not enabled.
+Trigger:
+    Periodic
+Reports on:
+    AWS::Elasticsearch::Domain
+Rule Parameters:
+    None
+
+Scenarios:
+  Scenario: 1
+     Given: No Amazon Elasticsearch Service (Amazon ES) domains are present in a region for an account.
+     Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: At least one Elasticsearch Service Domain is present
+     Given: Enabled field is true in EncryptionAtRestOptions
+     Then: Return COMPLIANT
+  Scenario: 3
+     Given: At least one Elasticsearch Service Domain is present
+     Given: Enabled field is false in EncryptionAtRestOptions
+     Then: Return NON_COMPLIANT
+"""
+
+import json
+import sys
+import time
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Elasticsearch::Domain'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+PAUSE_TO_AVOID_THROTTLE_SECONDS = 4
+
+def get_all_domain_details(es_client, es_domains):
+    es_domain_list_details = []
+    es_domains_names_only = []
+    for es_domain in es_domains:
+        es_domains_names_only.append(es_domain['DomainName'])
+    while es_domains_names_only:
+        time.sleep(PAUSE_TO_AVOID_THROTTLE_SECONDS)
+        domain_details = es_client.describe_elasticsearch_domains(DomainNames=es_domains_names_only[:5])['DomainStatusList']
+        es_domain_list_details += domain_details
+        del es_domains_names_only[:5]
+    return es_domain_list_details
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    es_client = get_client('es', event)
+    es_domain_list = es_client.list_domain_names()['DomainNames']
+    evaluations = []
+
+    if not es_domain_list:
+        return 'NOT_APPLICABLE'
+
+    es_domain_list_details = get_all_domain_details(es_client, es_domain_list)
+
+    for es_domain_details in es_domain_list_details:
+        if es_domain_details['EncryptionAtRestOptions']['Enabled']:
+            evaluations.append(build_evaluation(es_domain_details['DomainName'], 'COMPLIANT', event))
+        else:
+            evaluations.append(build_evaluation(es_domain_details['DomainName'], 'NON_COMPLIANT', event, annotation='This Amazon Elasticsearch domain does not have the encryption of data at rest as enabled'))

```

Comment:
`This Amazon Elasticsearch domain does not have the encryption of data at rest as enabled`
to
`This Amazon Elasticsearch domain is not encrypted at rest.`